src/python/config/ModuleDefine.cmake

```diff
@@ -100,8 +103,14 @@ function(PYIMATH_DEFINE_MODULE modname)
   # NB: make this one last so we can cheat and add the python and boost
   # libs as private deps at the end regardless of whether it was provided
   list(APPEND libarglist PRIVATE_DEPS ${PYIMATH_CURMOD_PRIVATE_DEPS})
+
+  set(optional_boost "")
+  if (TARGET Boost::${PYIMATH_BOOST_PY_COMPONENT})
+    set(optional_boost Boost::${PYIMATH_BOOST_PY_COMPONENT})
+  endif()
```

Comment:
I think the modern cmake approach is to use a generator expression like

    $<TARGET_NAME_IF_EXISTS:Boost::${PYIMATH_BOOST_PY_COMPONENT}>

and then you don't need the extra optional_boost variable or this `if` statement.